docs/verification.md

```diff
@@ -0,0 +1,71 @@
+verifying the results. Here’s an overview of the key steps involved:
+
+- Verifying the circuit we are securing does not include security holes;
+- Verifying the transcript matches the circuit;
+- Verifying contributors' attestations match the transcript;
+
+## Verifying the Circuit
+Confirming the integrity and security of the [EthStorage blob poseidon circuit](https://github.com/ethstorage/zk-decoder/blob/main/circom/circuits/blob_poseidon_2.circom) 
+by reviewing our circom code, which is open-source and available on Github.
+
+## Verifying the Transcript Matches the Circuit
+Our ceremony transcript is included in the zkey published on ceremony.ethstorage.io in the "Download ZKey" tab of the 
+[EthStorage Circom page](https://ceremony.ethstorage.io/projects/EthStorage%20circom). You can verify it matches the 
+circuit using Circom and snarkjs:
+
+- Install [Circom](https://docs.circom.io/getting-started/installation/) and [snarkjs](https://github.com/iden3/snarkjs).
+- Clone circom code https://github.com/ethstorage/zk-decoder.
+- Download the Powers of Tau (ptau) file we used for our ceremony; we used the Hermez rollup with 2^20 powers of tau, 
+which is linked in [the snarkjs readme](https://github.com/iden3/snarkjs) or available directly 
+[here](https://storage.googleapis.com/zkevm/ptau/powersOfTau28_hez_final_20.ptau).
+- Generate the `r1cs` file from the `circom` source files, by running `circom blob_poseidon_2.circom --r1cs` in the 
+directory `zk-decoder/circom/circuits`. The SHA-256 hash of this file (i.e. as computed by `sha256sum blob_poseidon_2.r1cs`) 
+is `ba17a94a5edcc1d0b0ad10990e0aea4894f080ce571b0db79fb1c3a9c4a74a0d`
+- Use snarkjs to verify that the transcript matches this circuit and powers of tau, by running 
+`snarkjs zkey verify blob_poseidon_2.r1cs powersOfTau28_hez_final_20.ptau blob_poseidon_2_final.zkey`. You should see 
+a list of contribution hashes (attestations) followed by the message snarkJS: `ZKey Ok!`.
+
+
+## Verifying Contributor Attestations Match the Transcript
+
+Each contributor authorized our ceremony to publish an attestation to their GitHub account. (In a Gist entitled 
+"`ethstorage-circom_attestation.log`") These attestations match the attestations published in the transcript. 
+This proves which GitHub user made each contribution to the ceremony. You can verify attestations by looking them 
+up both in the transcript and on GitHub:
+
+In the transcript (generated by snarkjs in the previous step), you will see that each contribution is preceded 
+by the phrase contribution `#[number] [username]-[number]`. That username is the GitHub username.
+
+On GitHub, you can run a search to find an attestation Gist on gist.github.com. For example, if the username is 
+`somecontributor`, then you could search for filename:`"ethstorage-circom_attestation.log" user:somecontributor`.
+
+To verify an attestation, confirm that the hash in the attestation in the transcript matches the hash in the 
```

Comment:
Can not find Ping-Ke's first contribution on the gist

Reply:
That is because it is create by ceremony setup

Reply:
But it should be available on the GitHub Gist, right?